src/Authoring/WinRT.SourceGenerator/Helper.cs

```diff
@@ -603,7 +603,14 @@ public static string GetMarshalerClass(string type, string abiType, TypeKind kin
                 }
                 else
                 {
-                    return "global::ABI." + type;
+                    if (isArray)
+                    {
+                        return $$"""MarshalNonBlittable<{{type}}>""";

```

Comment:
Just so I understand, this doesn't seem to be specific to `bool` values, so won't this change cause _all_ arrays of value types that are not the same as their ABI types to use `MarshalNonBlittable`? ...Is this intended? 🤔

Reply:
Yea I was making it match what code_writers did: https://github.com/microsoft/CsWinRT/blob/6257088ab16ce74bf04726c5062c606d77c0ca8f/src/cswinrt/code_writers.h#L4370